From f9a24eb07e1b316142852d4bbe5736aeea1ddabe Mon Sep 17 00:00:00 2001
From: Caleb Maclennan <caleb@alerque.com>
Date: Thu, 26 Oct 2023 00:03:08 +0300
Subject: [PATCH] Patch config files with Arch Linux specific locations

Arch uses upstream's default config files as a base for its configs.
But directory structure at Arch is completely different from the default
one specified by gitlab project. We used to have a lot of seds
expressions to adjust the files but as complexity of configs grew 'sed'
makes it easy to miss a changed/added option.

Track set of diffs as a patch. If upstream modifies config file then it
will cause a conflict that needs to be reviewed manually.
---
 config/database.yml.postgresql     | 79 +-----------------------------
 config/gitlab.yml.example          | 72 +++++++++++++--------------
 config/initializers/1_settings.rb  |  6 +--
 config/puma.example.development.rb | 16 +++---
 config/puma.rb.example             | 16 +++---
 5 files changed, 57 insertions(+), 132 deletions(-)

diff --git a/config/database.yml.postgresql b/config/database.yml.postgresql
index da9f458ff09..78b0dedae79 100644
--- a/config/database.yml.postgresql
+++ b/config/database.yml.postgresql
@@ -6,7 +6,7 @@ production:
     adapter: postgresql
     encoding: unicode
     database: gitlabhq_production
-    username: git
+    username: gitlab
     password: "secure password"
     host: localhost
     # load_balancing:
@@ -18,21 +18,6 @@ production:
     #     port: 8600
     #     record: secondary.postgresql.service.consul
     #     interval: 300
-  ci:
-    adapter: postgresql
-    encoding: unicode
-    database: gitlabhq_production
-    database_tasks: false
-    username: git
-    password: "secure password"
-    host: localhost
-  geo:
-    adapter: postgresql
-    encoding: unicode
-    database: gitlabhq_geo_production
-    username: git
-    password: "secure password"
-    host: localhost
 
 #
 # Development specific
@@ -47,23 +32,6 @@ development:
     host: localhost
     variables:
       statement_timeout: 15s
-  ci:
-    adapter: postgresql
-    encoding: unicode
-    database: gitlabhq_development
-    database_tasks: false
-    username: postgres
-    password: "secure password"
-    host: localhost
-    variables:
-      statement_timeout: 15s
-  geo:
-    adapter: postgresql
-    encoding: unicode
-    database: gitlabhq_geo_development
-    username: postgres
-    password: "secure password"
-    host: localhost
 
 #
 # Staging specific
@@ -73,22 +41,7 @@ staging:
     adapter: postgresql
     encoding: unicode
     database: gitlabhq_staging
-    username: git
-    password: "secure password"
-    host: localhost
-  ci:
-    adapter: postgresql
-    encoding: unicode
-    database: gitlabhq_staging
-    database_tasks: false
-    username: git
-    password: "secure password"
-    host: localhost
-  geo:
-    adapter: postgresql
-    encoding: unicode
-    database: gitlabhq_geo_staging
-    username: git
+    username: gitlab
     password: "secure password"
     host: localhost
 
@@ -107,31 +60,3 @@ test: &test
     reaping_frequency: nil
     variables:
       statement_timeout: 15s
-  ci:
-    adapter: postgresql
-    encoding: unicode
-    database: gitlabhq_test
-    database_tasks: false
-    username: postgres
-    password:
-    host: localhost
-    prepared_statements: false
-    reaping_frequency: nil
-    variables:
-      statement_timeout: 15s
-  geo:
-    adapter: postgresql
-    encoding: unicode
-    database: gitlabhq_geo_test
-    username: postgres
-    password:
-    host: localhost
-    reaping_frequency: nil
-  embedding:
-    adapter: postgresql
-    encoding: unicode
-    database: gitlabhq_embedding_test
-    username: postgres
-    password:
-    host: localhost
-    reaping_frequency: nil
diff --git a/config/gitlab.yml.example b/config/gitlab.yml.example
index e1163684755..a727db86c39 100644
--- a/config/gitlab.yml.example
+++ b/config/gitlab.yml.example
@@ -91,7 +91,7 @@ production: &base
       #- 2001:0db8::/32
 
     # Uncomment and customize if you can't use the default user to run GitLab (default: 'git')
-    # user: git
+    user: gitlab
 
     ## Date & Time settings
     # Uncomment and customize if you want to change the default time zone of GitLab application.
@@ -111,15 +111,15 @@ production: &base
       # enabled: false
       # S/MIME private key file in PEM format, unencrypted
       # Default is '.gitlab_smime_key' relative to Rails.root (i.e. root of the GitLab app).
-      # key_file: /home/git/gitlab/.gitlab_smime_key
+      # key_file: /var/lib/gitlab/.gitlab_smime_key
       # S/MIME public certificate key in PEM format, will be attached to signed messages
       # Default is '.gitlab_smime_cert' relative to Rails.root (i.e. root of the GitLab app).
-      # cert_file: /home/git/gitlab/.gitlab_smime_cert
+      # cert_file: /var/lib/gitlab/.gitlab_smime_cert
       # S/MIME extra CA public certificates in PEM format, will be attached to signed messages
       # Optional
-      # ca_certs_file: /home/git/gitlab/.gitlab_smime_ca_certs
+      # ca_certs_file: /var/lib/gitlab/.gitlab_smime_ca_certs
 
-    # Email server smtp settings are in config/initializers/smtp_settings.rb.sample
+    # Email server smtp settings are in /etc/webapps/gitlab/smtp_settings.rb
     # File location to read encrypted SMTP secrets from
     # email_smtp_secret_file: /mnt/gitlab/smtp.yaml.enc # Default: shared/encrypted_settings/smtp.yaml.enc
 
@@ -184,7 +184,7 @@ production: &base
     ## Repository downloads directory
     # When a user clicks e.g. 'Download zip' on a project, a temporary zip file is created in the following directory.
     # The default is 'shared/cache/archive/' relative to the root of the Rails app.
-    # repository_downloads_path: shared/cache/archive/
+    repository_downloads_path: /var/lib/gitlab/shared/cache/archive/
 
     ## Impersonation settings
     impersonation_enabled: true
@@ -251,7 +251,7 @@ production: &base
     # Since `mail_room` is run independently of Rails, an absolute path is preferred.
     # The default is 'log/mail_room_json.log' relative to the root of the Rails app.
     #
-    # log_path: log/mail_room_json.log
+    log_path: /var/log/gitlab/mail_room_json.log
 
     # If you are using Microsoft Graph instead of IMAP, set this to false to retain
     # messages in the inbox since deleted messages are auto-expunged after some time.
@@ -286,7 +286,7 @@ production: &base
     # contains the shared secret key for verifying access for mailroom's
     # incoming_email.
     # Default is '.gitlab_mailroom_secret' relative to Rails.root (i.e. root of the GitLab app).
-    # secret_file: /home/git/gitlab/.gitlab_mailroom_secret
+    # secret_file: /var/lib/gitlab/.gitlab_mailroom_secret
 
     # File location to read encrypted incoming email secrets from
     # encrypted_secret_file: /mnt/gitlab/smtp.yaml.enc
@@ -326,7 +326,7 @@ production: &base
   artifacts:
     enabled: true
     # The location where build artifacts are stored (default: shared/artifacts).
-    # path: shared/artifacts
+    path: /var/lib/gitlab/shared/artifacts
     # object_store:
     #   enabled: false
     #   remote_directory: artifacts # The bucket name
@@ -345,7 +345,7 @@ production: &base
     # be stored on disk, or in object storage
     enabled: false
     # The location where external diffs are stored (default: shared/lfs-external-diffs).
-    # storage_path: shared/external-diffs
+    storage_path: /var/lib/gitlab/shared/external-diffs
     # object_store:
     #   enabled: false
     #   remote_directory: external-diffs
@@ -360,7 +360,7 @@ production: &base
   lfs:
     enabled: true
     # The location where LFS objects are stored (default: shared/lfs-objects).
-    # storage_path: shared/lfs-objects
+    storage_path: /var/lib/gitlab/shared/lfs-objects
     object_store:
       enabled: false
       remote_directory: lfs-objects # Bucket name
@@ -379,7 +379,7 @@ production: &base
   ## Uploads (attachments, avatars, etc...)
   uploads:
     # The location where uploads objects are stored (default: public/).
-    # storage_path: public/
+    storage_path: /var/lib/gitlab/public/
     # base_dir: uploads/-/system
     object_store:
       enabled: false
@@ -400,7 +400,7 @@ production: &base
     enabled: true
     dpkg_deb_path: /usr/bin/dpkg-deb
     # The location where build packages are stored (default: shared/packages).
-    # storage_path: shared/packages
+    storage_path: /var/lib/gitlab/shared/packages
     object_store:
       enabled: false
       remote_directory: packages # The bucket name
@@ -419,7 +419,7 @@ production: &base
   dependency_proxy:
     enabled: true
     # The location where build packages are stored (default: shared/dependency_proxy).
-    # storage_path: shared/dependency_proxy
+    storage_path: /var/lib/gitlab/shared/dependency_proxy
     object_store:
       enabled: false
       remote_directory: dependency_proxy # The bucket name
@@ -438,7 +438,7 @@ production: &base
   terraform_state:
     enabled: true
     # The location where Terraform state files are stored (default: shared/terraform_state).
-    # storage_path: shared/terraform_state
+    storage_path: /var/lib/gitlab/shared/terraform_state
     object_store:
       enabled: false
       remote_directory: terraform # The bucket name
@@ -455,7 +455,7 @@ production: &base
   ## CI Secure Files
   ci_secure_files:
     enabled: true
-    # storage_path: shared/ci_secure_files
+    storage_path: /var/lib/gitlab/shared/ci_secure_files
     object_store:
       enabled: false
       remote_directory: ci-secure-files # The bucket name
@@ -474,7 +474,7 @@ production: &base
     enabled: false
     access_control: false
     # The location where pages are stored (default: shared/pages).
-    # path: shared/pages
+    path: /var/lib/gitlab/shared/pages
 
     # The domain under which the pages are served:
     # http://group.example.com/project
@@ -488,7 +488,7 @@ production: &base
 
     # File that contains the shared secret key for verifying access for gitlab-pages.
     # Default is '.gitlab_pages_secret' relative to Rails.root (i.e. root of the GitLab app).
-    # secret_file: /home/git/gitlab/.gitlab_pages_secret
+    secret_file: /var/lib/gitlab/.gitlab_pages_secret
     object_store:
       enabled: false
       remote_directory: pages # The bucket name
@@ -684,7 +684,7 @@ production: &base
     # port: 5005
     # api_url: http://localhost:5000/ # internal address to the registry, will be used by GitLab to directly communicate with API
     # key: config/registry.key
-    # path: shared/registry
+    path: /var/lib/gitlab/shared/registry
     # issuer: gitlab-issuer
     # notification_secret: '' # only set it when you use Geo replication feature without built-in Registry
 
@@ -735,7 +735,7 @@ production: &base
     # Default project notifications settings:
 
     # The location where build traces are stored (default: builds/). Relative paths are relative to Rails.root
-    # builds_path: builds/
+    builds_path: /var/lib/gitlab/builds/
 
   #
   # 3. Auth settings
@@ -1188,7 +1188,7 @@ production: &base
 
   # Shared file storage settings
   shared:
-    # path: /mnt/gitlab # Default: shared
+    path: /var/lib/gitlab/shared # Default: shared
 
   # Encrypted Settings configuration
   encrypted_settings:
@@ -1209,13 +1209,13 @@ production: &base
   repositories:
     storages: # You must have at least a `default` storage path.
       default:
-        gitaly_address: unix:/home/git/gitlab/tmp/sockets/private/gitaly.socket # TCP connections are supported too (e.g. tcp://host:port). TLS connections are also supported using the system certificate pool (eg: tls://host:port).
+        gitaly_address: unix:/run/gitlab/gitlab-gitaly.socket # TCP connections are supported too (e.g. tcp://host:port). TLS connections are also supported using the system certificate pool (eg: tls://host:port).
         # gitaly_token: 'special token' # Optional: override global gitaly.token for this storage.
 
   ## Backup settings
   backup:
-    path: "tmp/backups"   # Relative paths are relative to Rails.root (default: tmp/backups/)
-    # gitaly_backup_path: /home/git/gitaly/_build/bin/gitaly-backup # Path of the gitaly-backup binary (default: searches $PATH)
+    path: "/var/lib/gitlab/backups"   # Relative paths are relative to Rails.root (default: tmp/backups/)
+    # gitaly_backup_path: /usr/bin/gitaly-backup # Path of the gitaly-backup binary (default: searches $PATH)
     # archive_permissions: 0640 # Permissions for the resulting backup.tar file (default: 0600)
     # keep_time: 604800   # default: 0 (forever) (in seconds)
     # pg_schema: public     # default: nil, it means that all schemas will be backed up
@@ -1250,12 +1250,12 @@ production: &base
 
   ## GitLab Shell settings
   gitlab_shell:
-    path: /home/git/gitlab-shell/
-    authorized_keys_file: /home/git/.ssh/authorized_keys
+    path: /usr/share/webapps/gitlab-shell/
+    authorized_keys_file: /var/lib/gitlab/.ssh/authorized_keys
 
     # File that contains the secret key for verifying access for gitlab-shell.
     # Default is '.gitlab_shell_secret' relative to Rails.root (i.e. root of the GitLab app).
-    # secret_file: /home/git/gitlab/.gitlab_shell_secret
+    # secret_file: /var/lib/gitlab/.gitlab_shell_secret
 
     # Git over HTTP
     upload_pack: true
@@ -1270,14 +1270,14 @@ production: &base
   workhorse:
     # File that contains the secret key for verifying access for gitlab-workhorse.
     # Default is '.gitlab_workhorse_secret' relative to Rails.root (i.e. root of the GitLab app).
-    # secret_file: /home/git/gitlab/.gitlab_workhorse_secret
+    # secret_file: /var/lib/gitlab/.gitlab_workhorse_secret
 
   topology_service:
     # enabled: false
     # address: topology-service.gitlab.example.com:443
-    # ca_file: /home/git/gitlab/config/topology-service-ca.pem
-    # certificate_file: /home/git/gitlab/config/topology-service-cert.pem
-    # private_key_file: /home/git/gitlab/config/topology-service-key.pem
+    # ca_file: /var/lib/gitlab/config/topology-service-ca.pem
+    # certificate_file: /var/lib/gitlab/config/topology-service-cert.pem
+    # private_key_file: /var/lib/gitlab/config/topology-service-key.pem
 
   cell:
     # name: cell-1
@@ -1286,7 +1286,7 @@ production: &base
     # enabled: true
     # File that contains the secret key for verifying access for gitlab-kas.
     # Default is '.gitlab_kas_secret' relative to Rails.root (i.e. root of the GitLab app).
-    # secret_file: /home/git/gitlab/.gitlab_kas_secret
+    # secret_file: /var/lib/gitlab/.gitlab_kas_secret
 
     # The URL to the external KAS API (used by the Kubernetes agents)
     # external_url: wss://kas.example.com
@@ -1300,17 +1300,17 @@ production: &base
   suggested_reviewers:
     # File that contains the secret key for verifying access to GitLab internal API for Suggested Reviewers.
     # Default is '.gitlab_suggested_reviewers_secret' relative to Rails.root (i.e. root of the GitLab app).
-    # secret_file: /home/git/gitlab/.gitlab_suggested_reviewers_secret
+    # secret_file: /var/lib/gitlab/.gitlab_suggested_reviewers_secret
 
   zoekt:
     # Files that contain username and password for basic auth for Zoekt
     # Default is '.gitlab_zoekt_username' and '.gitlab_zoekt_password' in Rails.root
-    # username_file: /home/git/gitlab/.gitlab_zoekt_username
-    # password_file: /home/git/gitlab/.gitlab_zoekt_password
+    # username_file: /var/lib/gitlab/.gitlab_zoekt_username
+    # password_file: /var/lib/gitlab/.gitlab_zoekt_password
 
   ## GitLab Elasticsearch settings
   elasticsearch:
-    indexer_path: /home/git/gitlab-elasticsearch-indexer/
+    indexer_path: /var/lib/gitlab/elasticsearch-indexer/
 
   ## Git settings
   # CAUTION!
diff --git a/config/initializers/1_settings.rb b/config/initializers/1_settings.rb
index 5ac698e2b6a..27d85e7b70f 100644
--- a/config/initializers/1_settings.rb
+++ b/config/initializers/1_settings.rb
@@ -976,9 +976,9 @@
 Settings['topology_service'] ||= {}
 Settings.topology_service['enabled'] ||= false
 Settings.topology_service['address'] ||= 'topology-service.gitlab.example.com:443'
-Settings.topology_service['ca_file'] ||= '/home/git/gitlab/config/topology-service-ca.pem'
-Settings.topology_service['certificate_file'] ||= '/home/git/gitlab/config/topology-service-cert.pem'
-Settings.topology_service['private_key_file'] ||= '/home/git/gitlab/config/topology-service-key.pem'
+Settings.topology_service['ca_file'] ||= '/var/lib/gitlab/config/topology-service-ca.pem'
+Settings.topology_service['certificate_file'] ||= '/var/lib/gitlab/config/topology-service-cert.pem'
+Settings.topology_service['private_key_file'] ||= '/var/lib/gitlab/config/topology-service-key.pem'
 
 #
 # Cells
diff --git a/config/puma.example.development.rb b/config/puma.example.development.rb
index e7dacc70c2e..5f64b57afea 100644
--- a/config/puma.example.development.rb
+++ b/config/puma.example.development.rb
@@ -2,7 +2,7 @@
 
 # -----------------------------------------------------------------------
 # This file is used by the GDK to generate a default config/puma.rb file
-# Note that `/home/git` will be substituted for the actual GDK root
+# Note that `/var/lib` will be substituted for the actual GDK root
 # directory when this file is generated
 # -----------------------------------------------------------------------
 
@@ -11,15 +11,15 @@
 # The default is "config.ru".
 #
 rackup 'config.ru'
-pidfile '/home/git/gitlab/tmp/pids/puma.pid'
-state_path '/home/git/gitlab/tmp/pids/puma.state'
+pidfile '/var/lib/gitlab/tmp/pids/puma.pid'
+state_path '/var/lib/gitlab/tmp/pids/puma.state'
 
 ## Uncomment the lines if you would like to write puma stdout & stderr streams
 ## to a different location than rails logs.
 ## When using GitLab Development Kit, by default, these logs will be consumed
 ## by runit and can be accessed using `gdk tail rails-web`
-# stdout_redirect '/home/git/gitlab/log/puma.stdout.log',
-#  '/home/git/gitlab/log/puma.stderr.log',
+# stdout_redirect '/var/lib/gitlab/log/puma.stdout.log',
+#  '/var/lib/gitlab/log/puma.stderr.log',
 #  true
 
 # Configure "min" to be the minimum number of threads to use to answer
@@ -41,11 +41,11 @@
 
 # Bind the server to "url". "tcp://", "unix://" and "ssl://" are the only
 # accepted protocols.
-bind 'unix:///home/git/gitlab.socket'
+bind 'unix:///var/lib/gitlab.socket'
 
 workers 2
 
-require_relative "/home/git/gitlab/lib/gitlab/cluster/lifecycle_events"
+require_relative "/var/lib/gitlab/lib/gitlab/cluster/lifecycle_events"
 
 on_restart do
   # Signal application hooks that we're about to restart
@@ -85,7 +85,7 @@
 wait_for_less_busy_worker ENV.fetch('PUMA_WAIT_FOR_LESS_BUSY_WORKER', 0.001).to_f
 
 # Use json formatter
-require_relative "/home/git/gitlab/lib/gitlab/puma_logging/json_formatter"
+require_relative "/var/lib/gitlab/lib/gitlab/puma_logging/json_formatter"
 
 json_formatter = Gitlab::PumaLogging::JSONFormatter.new
 log_formatter do |str|
diff --git a/config/puma.rb.example b/config/puma.rb.example
index 65c817bb2d8..2d6aa7c6482 100644
--- a/config/puma.rb.example
+++ b/config/puma.rb.example
@@ -5,11 +5,11 @@
 # The default is "config.ru".
 #
 rackup 'config.ru'
-pidfile '/home/git/gitlab/tmp/pids/puma.pid'
-state_path '/home/git/gitlab/tmp/pids/puma.state'
+pidfile '/run/gitlab/puma.pid'
+state_path '/run/gitlab/puma.state'
 
-stdout_redirect '/home/git/gitlab/log/puma.stdout.log',
-  '/home/git/gitlab/log/puma.stderr.log',
+stdout_redirect '/var/log/gitlab/puma.stdout.log',
+  '/var/log/gitlab/puma.stderr.log',
   true
 
 # Configure "min" to be the minimum number of threads to use to answer
@@ -31,11 +31,11 @@ queue_requests false
 
 # Bind the server to "url". "tcp://", "unix://" and "ssl://" are the only
 # accepted protocols.
-bind 'unix:///home/git/gitlab/tmp/sockets/gitlab.socket'
+bind 'unix:///run/gitlab/gitlab.socket'
 
 workers 3
 
-require_relative "/home/git/gitlab/lib/gitlab/cluster/lifecycle_events"
+require_relative "/usr/share/webapps/gitlab/lib/gitlab/cluster/lifecycle_events"
 
 on_restart do
   # Signal application hooks that we're about to restart
@@ -74,14 +74,14 @@ worker_timeout 60
 wait_for_less_busy_worker ENV.fetch('PUMA_WAIT_FOR_LESS_BUSY_WORKER', 0.001).to_f
 
 # Use json formatter
-require_relative "/home/git/gitlab/lib/gitlab/puma_logging/json_formatter"
+require_relative "/usr/share/webapps/gitlab/lib/gitlab/puma_logging/json_formatter"
 
 json_formatter = Gitlab::PumaLogging::JSONFormatter.new
 log_formatter do |str|
   json_formatter.call(str)
 end
 
-require_relative "/home/git/gitlab/lib/gitlab/puma/error_handler"
+require_relative "/usr/share/webapps/gitlab/lib/gitlab/puma/error_handler"
 
 error_handler = Gitlab::Puma::ErrorHandler.new(ENV['RAILS_ENV'] == 'production')
 
-- 
2.45.2

